### ISSUE TITLE
ignoreInDest options doesn't allow negation

### ISSUE THREAD

**Issue by guncha** [tomusdrw/grunt-sync#36]
After looking at the code briefly, it looks like any `!**/*.ext` pattern doesn't work in the `ignoreInDest` option due to the pattern being combined with the `dest` directory. This makes it impossible to have `grunt-sync` only work on, say, coffee files. Thoughts?

**Comment by tomusdrw** [comment]
If you want to work only on coffee files then you just need to configure the source to be coffee files:
```javascript
files: [{
  cwd: 'src',
  src: [
    '**/*.coffee' /* include only coffee files in `src` */
  ],
  dest: 'bin',
}]
```

`ignoreInDest` is only used when the plugin is removing the files in `dest` that are not found in `src`. You don't want `grunt-sync` to delete some files that might come from different source.

So if yours `*.coffee` files are not synchronized by `grunt-sync` you can ignore them - but then you don't need negative pattern. It just means "`ignore` `**/*.coffee` files `inDest`" (don't remove them even though they are not found in `src`).

Does it make sense? Maybe I need to update readme to describe some common use cases.

**Comment by guncha** [comment]
Thanks for getting back. My issue is that `grunt-sync` wants to remove all other files that are in the directory, files that are not matching `*.coffee` like a bunch of javascripts in subdirectories.

Here's my config:

```coffee
sync:
  coffee:
    files: [{
      expand: true
      cwd: '<%= yeoman.app %>/scripts'
      dest: '.tmp/scripts'
      src: '**/*.coffee'
    }]
    verbose: true
    updateAndDelete: true
    # NOTE: this is a little fragile as grunt-sync doesn't allow for negation patterns
    # so if you see your files mysteriously disappearing from .tmp/scripts, this is it.
    # ignoreInDest: ["{vendor,templates}/**", "**/*.{js,json}"]
```

I'm working around this behavior using the uncommented `ignoreInDest` option, but it's clearly fragile and not to be trusted. Is there a better way to have `grunt-sync` leave other files that it didn't put there alone?

From `package.json`: `"grunt-sync": "^0.3.0",`

**Comment by tomusdrw** [comment]
I think I've made a fix for that - you can find it in this commit:
https://github.com/tomusdrw/grunt-sync/commit/4b3f9b69c34e1c05d115a922e8a551f0aefdc4db

It should now allow to apply negated patterns (in fact any `glob` patterns) to ignoreInDest. 
Let me know if this helps and then I will release version `0.4.0`

**Comment by guncha** [comment]
Works like a charm! :+1:

**Comment by tomusdrw** [comment]
Released as `0.4.0`

**Comment by benoz23** [comment]
Then i can't understand where i'm going wrong.
My `sync:html` task is supposed to work on all html files, ignoring all but those files:
```CoffeeScript
sync: {

  ...

  html: {
    files: [{
      cwd: 'dev/',
      src: ['**/*.html'],
      dest: 'dist/'
    }],
    ignoreInDest: [
      '**',
      '!**/*.html'
    ],
    updateAndDelete: true
  },

  ...

}
```
This seems to actually ignore everything, as if it neglects the `'!**/*.html'` instruction.

What am i missing?

`"grunt-sync": "^0.5.2"`
